src/osd_field.c

```diff
@@ -0,0 +1,359 @@
+// SPDX-License-Identifier: GPL-2.0-only
+#include <assert.h>
+#include <wlr/util/log.h>
+#include "common/mem.h"
+#include "config/rcxml.h"
+#include "view.h"
+#include "workspaces.h"
+#include "labwc.h"
+#include "osd.h"
+
+/* includes '%', terminating 's' and NULL byte, 8 is enough for %-9999W */
+#define LAB_FIELD_SINGLE_FMT_MAX_LEN 8
+static_assert(LAB_FIELD_SINGLE_FMT_MAX_LEN <= 255, "fmt_len is a unsigned char");
+
+/* forward declares */
+typedef void field_set_fn_t(struct buf *buf, struct view *view, const char *format);
+struct field_handlers {
+	const char fmt_char;
```

Comment:
Suggest sticking with `printf` language, so better to call this `conversion_specifier`.

Reply:
field_set_fn_t becomes field_conversion_type

field_handlers becomes field_converter